src/SFA.DAS.Payments.DataLocks.DataLockService/DataLockService.cs

```diff
@@ -41,15 +46,34 @@ public class DataLockService : Actor, IDataLockService
             this.paymentLogger = paymentLogger;
             this.apprenticeshipRepository = apprenticeshipRepository;
             this.apprenticeships = apprenticeships;
+            this.learnerMatcher = learnerMatcher;
+            this.courseValidator = courseValidator;
         }
 
         public async Task<DataLockEvent> HandleEarning(ApprenticeshipContractType1EarningEvent message, CancellationToken cancellationToken)
         {
-            var apprenticeshipsForUln = await apprenticeships.TryGet(message.Learner.Uln.ToString(), cancellationToken)
-                .ConfigureAwait(false);
-            var apprenticeship = apprenticeshipsForUln.Value.FirstOrDefault();
+            var learnerMatchResult = await learnerMatcher.MatchLearner(message.Ukprn);
+
+            if (learnerMatchResult.DataLockErrorCode.HasValue)
+            {
+                // TODO: return non-payable earning
+                return null;
```

Comment:
This introduces a `null` return in a `Task<T>` method, which will cause a NullReferenceException in the calling `await`.  

```suggestion
                return Task.FromResult<DataLockEvent>(null);
```

Reply:
These code changes are coming from PV2-763. Please make the comment on the PR.